localstack/services/dynamodb/dynamodb_listener.py

```diff
@@ -1020,12 +1021,21 @@ def update_global_table(data):
 # ---
 
 
-def is_index_query_valid(table_name: str, index_query_type: str) -> bool:
+def get_global_secondary_index(table_name, index_name):
     schema = SchemaExtractor.get_table_schema(table_name)
     for index in schema["Table"].get("GlobalSecondaryIndexes", []):
-        index_projection_type = index.get("Projection").get("ProjectionType")
-        if index_query_type == "ALL_ATTRIBUTES" and index_projection_type != "ALL":
-            return False
```

Comment:
This was returning False if _any_ index did not have a projection type of ALL, but only the index actually being queried should be checked.